### DISCUSSION TITLE
Receiving parameter value errors when doing emcee sampling

### DISCUSSION THREAD

**Discussion by hwillems2** [phoebe-project/phoebe2#936] [Questions]
After successfully running Nelder-Mead on my system, I am now trying to estimate the parameter uncertainties via emcee sampling. 
But now I got an error shown in the screenshot: 
![Screenshot (19)](https://github.com/user-attachments/assets/46c4e768-1236-4ba5-82c1-b49c279034ed)
It basically says the radius of a star is negative which is of course unphysical.

Relevant lines of code:
`logger = phoebe.logger('warning')
    
    b = phoebe.default_binary()
    b.set_value_all('atm', value='blackbody')
    b.set_value_all('gravb_bol', value=1)
    b.set_value_all('irrad_frac_refl_bol', value=1)
    b.set_value_all('ld_mode_bol', value='manual')
    b.set_value('ecc', 0.157)
    b.set_value('per0', 335)
    b.set_value('teff@primary',  22158.373900960192) #B2.5V star
    b.set_value('teff@secondary', 21107.551731764146)
    b.set_value('requiv@primary',  3.1924768444278953 )
    b.set_value('requiv@secondary', 2.8927745347890257) 
    b.set_value('incl@binary',  83.67567432598418)
    b.flip_constraint('asini@primary', solve_for='sma@binary')
    b.set_value('asini@primary', 12.45)

    b.set_value('period@binary',  4.39978499)
    b.set_value('q', 0.97) # fixed from the paper
    b.flip_constraint('t0_perpass@binary', solve_for='t0_supconj@binary')
    b.set_value('t0_perpass@binary', 2457302.31)

    b.set_value('vgamma', 250.16)
    
    lctimes = phoebe.linspace(0, 10, 1005)
    b.add_dataset('lc', compute_times=lctimes, dataset='lc01')
    b.set_value_all('ld_mode', 'manual')
    
    
    b = phoebe.default_binary()
    dir_path = os.path.dirname(os.path.realpath(__file__))
    times, fluxes, sigmas = np.loadtxt(f"{dir_path}/phot/I/LMC553.25.29442.dat", unpack=True)
    zero_point, flux=MagToFlux(fluxes, 0)
    b.add_dataset('lc',
                compute_phases = phoebe.linspace(0, 1, 40),
                times=times,
                fluxes=flux,
                sigmas=sigmas,
                dataset='lc02')
    b.set_value(qualifier='pblum_mode', context='dataset', dataset='lc02', value='dataset-scaled')
    b.set_value('period@binary',  4.39978499)
    

    
   
   
    
    b.set_value_all('atm', value='blackbody')
    b.set_value_all('gravb_bol', value=1)
    b.set_value_all('irrad_frac_refl_bol', value=1)
    b.set_value_all('ld_mode_bol', value='manual')
    b.set_value('ecc', 0.157)
    b.set_value('per0', 20)
    b.set_value('teff@primary',  22158.373900960192) #B2.5V star
    b.set_value('teff@secondary', 21107.551731764146)
    b.set_value('requiv@primary',  3.1924768444278953 )
    b.set_value('requiv@secondary', 2.8927745347890257) 
    b.set_value('incl@binary',  83.67567432598418)
    b.flip_constraint('asini@primary', solve_for='sma@binary')
    b.set_value('asini@primary',12.45)
    b.set_value('period@binary', 4.39978499) #4.40009 from LS
    b.set_value('q', 0.97) #fixed from the paper
    b.flip_constraint('t0_perpass@binary', solve_for='t0_supconj@binary')
    b.set_value('t0_perpass@binary',  2457300.11)
    b.set_value('vgamma', 250.16)
    b.set_value_all('ld_mode', 'manual')
    b.run_compute(compute='phoebe01', model='orig_model')
    
    _ = b.plot(x='phases', set_title='VFTS364 phases', show=True, save='.\emcee\VFTS364\emcee_VFTS364_phases.pdf')
    
    _ = b.plot(x='phases', set_title='VFTS364 residuals', y='residuals', show=True, save='.\emcee\VFTS364\emcee_VFTS364_resdiuals.pdf')
    
    b.add_solver('sampler.emcee', solver='emcee_solver')
    
    print(b.filter(solver='emcee_solver'))
    
    b.set_value('compute', solver='emcee_solver', value='phoebe01')
    
    b.add_distribution({"teff@primary@star@component": phoebe.gaussian_around(2000), 
            'teff@secondary@star@component': phoebe.gaussian_around(2000),
             'requiv@primary@star@component': phoebe.gaussian_around(1.5), 
             'requiv@secondary@star@component': phoebe.gaussian_around(1.5),
             # 'q': phoebe.gaussian_around(0.2), 
             'incl@binary@component': phoebe.gaussian_around(3)
             }, distribution='ball_around_guess')
    
    print(f'The distribution is: {b.get_distribution(distribution="ball_around_guess")}')
    b.set_value_all('ld_mode@lc02', 'manual')
    
    b.run_compute(compute='phoebe01', sample_from='ball_around_guess', 
                  sample_num=20, model='init_from_model')
   
    
    b.set_value('init_from', 'ball_around_guess')
    
    _ = b.plot_distribution_collection(distribution='ball_around_guess', parameters=['incl@binary@component'])
    
    b.set_value('priors','ball_around_guess')
    
    b.set_value('nwalkers', solver='emcee_solver', value=12)
    
    b.set_value('niters', solver='emcee_solver', value=2000)

    print(b.filter(qualifier='enabled', compute='phoebe01'))
    
    
    print(b.filter(qualifier='enabled', compute='phoebe01'))
    
    b.run_solver('emcee_solver', solution='emcee_sol')`

**Comment by kecnry** [comment]
These are just logger warnings, so are not a cause for major concern.  The emcee sampler is continuing, but just rejecting those samples because of the unphysical radius you mentioned.  The reason this is happening is because your initializing distribution are gaussians - which always have a chance of giving negative values since they have no cutoff - and are fairly broad which increases the probability of drawing negative values.  The sampler will eventually figure this out as long as it can find valid initial values for the walkers (in other words - as long as there is a large enough probability of finding valid values that it doesn't repeatedly keep drawing unphysical values).

If you are concerned that this is happening "too much" and causing the sampling process to slow down significantly, you can use a uniform distribution instead (or in combination with a gaussian) to ensure a hard cutoff on the original sampling.
